### PR TITLE
[3.x] Backport glTF module with scene export support

### PR DESCRIPTION
**Pull request by lyuma**
Updated version of gltf2 module by @fire including the preliminary GLTF Export support.

Optimistically includes a backport of #48915 as I needed this bugfix to test my content. It's the last commit in the chain and can be ignored if necessary.

Note that GLTF Export crashes when given complicated skinned meshes. This crash is also present in Master. I will need some additional help from fire to resolve export-related bugs.

In this version, I collapsed the commit history.

Supersedes #46783
Supersedes #49051

### CODE REVIEW

File: modules/gltf/doc_classes/EditorSceneImporterGLTF.xml
Comment (fire) [inline modules/gltf/doc_classes/EditorSceneImporterGLTF.xml:3]: I think it's time to document gltf for master and 3.x. Starting with a master pr, so it be a separate pr.

Comment (aaronfranke) [reply modules/gltf/doc_classes/EditorSceneImporterGLTF.xml:3]: I think that's a separate issue from the code backport itself. The lack of documentation shouldn't block this PR.

**Review by aaronfranke — APPROVED** [review]
I tested this on the TPS demo. Overall works great in my testing, with the same notes as I made [here](https://github.com/godotengine/godot/pull/46783#pullrequestreview-616862324) of slightly different behavior (technically it doesn't break compatibility as existing projects should keep working, but for best results user projects should be updated manually, otherwise there are duplicated imported meshes).

On the addition of the empty documentation files: Filling these in would be nice, but it's not a blocker for this PR in my opinion. It might make sense to get this merged in sooner rather than later, and the documentation can be written in another PR.

This will need approval from @fire before it can be merged.

**Review by fire — COMMENTED** [review]
We updated the master pr for skeletons bones that are also meshes. So merging this pr is linked to that being done.

I have prs incoming for export improvements

**Review by fire — APPROVED** [review]
Ready to merge because https://github.com/godotengine/godot/pull/49119 was merged.

Edited: 

Straightforward conflicts because of the changes to the older gltf importer. (49119)

**Comment by lyuma** [comment]
Updated the PR for godot 3.x and tested on my various gltf test scenes.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**
